semanticcpg/src/main/scala/io/shiftleft/semanticcpg/sarif/SarifSchema.scala

```diff
@@ -0,0 +1,315 @@
+package io.shiftleft.semanticcpg.sarif
+
+import org.json4s.{CustomSerializer, Extraction, Serializer}
+import org.slf4j.LoggerFactory
+
+import java.net.URI
+
+object SarifSchema {
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  /** Provides a basic Sarif trait under which possibly multiple defined schemata would be defined.
+    */
+  sealed trait Sarif {
+
+    /** @return
+      *   The SARIF format version of this log file.
+      */
+    def version: String
+
+    /** @return
+      *   The URI of the JSON schema corresponding to the version.
+      */
+    def `$schema`: String
```

Comment:
I guess this beautiful member name stems from the name it needs to have in the json? might be worth documenting the fact that a future refactoring must not change these names, unless the serialization is adapted.

Reply:
Yes this is required by the JSON schema. I can opt for a more sane name and adapt the serialization too.

Reply:
if it's straightforward, I'd be 👍🏻 
otherwise just add a comment